### PR TITLE
allow password to be referenced by secret

### LINKED ISSUE #61: [fluentd-elasticsearch] Auth password from secret
When deploying this chart for a cluster requiring authentication, the only way to provide the password is via helm values. It would be beneficial to additionally allow the password to be sourced from an existing secret, as so that this can be used with pre-existing secrets.

For our use case, this will enable us to use this chart as part of GitOps by utilising the Sealed Secrets operator

### PR DESCRIPTION
**Pull request by 0x4c6565**
# Which chart

fluentd-elasticsearch

# What this PR does / why we need it

Allow auth password to be specified by existing secret

# Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #61

# Special notes for your reviewer

# Checklist

- [x] [DCO](https://github.com/kokuwaio/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] All variables are documented in the charts README

### CODE REVIEW

**Comment by 0x4c6565** [comment]
DCO link dead

File: charts/fluentd-elasticsearch/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: fluentd-elasticsearch
-version: 11.8.0
+version: 11.8.1
```
Comment (monotek) [inline charts/fluentd-elasticsearch/Chart.yaml:3]: New feature should raise minor version

File: charts/fluentd-elasticsearch/values.yaml [hunk 2]
```diff
@@ -63,6 +63,9 @@ elasticsearch:
     enabled: false
     user: null
     password: null
+    passwordSecretKeyRef: null
```
Comment (monotek) [inline charts/fluentd-elasticsearch/values.yaml:66]: Imho this should look like:

```
    existingSecret:
      name: ""
      key: ""
```

In the deamonsert you can than just check for:

```
{{- if .Values.elasticsearch.auth.existingSecret.name }}
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by monotek** [comment]
> DCO link dead

fixed

File: charts/fluentd-elasticsearch/values.yaml (hunk 2 above)
Comment (0x4c6565) [reply charts/fluentd-elasticsearch/values.yaml]: Suggestion applied

File: charts/fluentd-elasticsearch/Chart.yaml (hunk 1 above)
Comment (0x4c6565) [reply charts/fluentd-elasticsearch/Chart.yaml]: Done

File: charts/fluentd-elasticsearch/values.yaml [hunk 3]
```diff
@@ -63,6 +63,9 @@ elasticsearch:
     enabled: false
     user: null
     password: null
+    existingSecret: null
```
Comment (monotek) [inline charts/fluentd-elasticsearch/values.yaml:66]: please remove null and comment empty values in.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/fluentd-elasticsearch/values.yaml (hunk 3 above)
Comment (0x4c6565) [reply charts/fluentd-elasticsearch/values.yaml]: Sorted

**Review by monotek — APPROVED** [review]

**Pull request merged**